ChildProject/pipelines/eafbuilder.py

```diff
@@ -1,28 +1,120 @@
+except ImportError:
+    # TODO: Perhaps add this as a dependency to the resources?
+    import importlib_resources as resources
 
 from ChildProject.projects import ChildProject
 from ChildProject.pipelines.pipeline import Pipeline
 
+def create_eaf(etf_path: str, id: str, output_dir: str,
+    timestamps_list: list,
+    eaf_type: str, contxt_on: int, contxt_off: int,
+    template: str):
+
+    print("ACLEW ID: ", id)
+
+    eaf = pympi.Elan.Eaf(etf_path)
+    ling_type = "transcription"
+    eaf.add_tier("code_"+eaf_type, ling=ling_type)
+    eaf.add_tier("context_"+eaf_type, ling=ling_type)
+    eaf.add_tier("code_num_"+eaf_type, ling=ling_type)
+
+    for i, ts in enumerate(timestamps_list):
+        print("Creating eaf code segment # ", i+1)
+        print("enumerate makes: ", i, ts)
+        whole_region_onset = ts[0]
+        whole_region_offset = ts[1]
+
+        context_onset = int(whole_region_onset) - contxt_on
+        context_offset = int(whole_region_offset) + contxt_off
+
+        if context_onset < 0:
+            context_onset = 0.0
+        
+        codeNumVal = eaf_type + str(i+1)
+        eaf.add_annotation("code_"+eaf_type, whole_region_onset, whole_region_offset)
+        eaf.add_annotation("code_num_"+eaf_type, whole_region_onset, whole_region_offset, value=codeNumVal)
+        eaf.add_annotation("context_"+eaf_type, context_onset, context_offset)
+
+    os.makedirs(output_dir, exist_ok = True)
+    eaf.to_file(os.path.join(output_dir, "{}.eaf".format(id)))
+    for i in eaf.get_tier_names():
+        print(i,":",eaf.get_annotation_data_for_tier(i))
+
+    return eaf
+
 class EafBuilderPipeline(Pipeline):
     def __init__(self):
         pass
                 
-    def run(self, path: str, destination: str, segments: str,
-        type: str, template: str,
-        context_onset: float, context_offset: float,
+    def run(self, destination: str, segments: str,
+        eaf_type: str, template: str,
+        context_onset: int = 0, context_offset: int = 0,
         **kwargs):
+        """generate .eaf templates based on intervals to code
+
+        :param path: project path
+        :type path: str
+        :param destination: eaf destination
+        :type destination: str
+        :param segments: path to the input segments dataframe
+        :type segments: str
+        :param eaf_type: eaf-type [random, periodic]
+        :type eaf_type: str
+        :param template: name of the template to use (basic, native, or non-native)
+        :type template: str
+        :param context_onset: context onset and segment offset difference in milliseconds, 0 for no introductory context
+        :type context_onset: float
+        :param context_offset: context offset and segment offset difference in milliseconds, 0 for no outro context
+        :type context_offset: float
+        """
+
+        # TODO: Make sure etf file paths are approprite and robust. 
+        etf_path = "{}.etf".format(template)
```

Comment:
these are not always provided, right? why do we want them?

Reply:
Right now we provide 3 templates with our package (so the users don't need their own), and only these can be used.

However, we could allow users to use their own, custom templates as well.

Or, we could remove the default templates from our package, and oblige users to manager their own templates.

What do you think is better ?

Reply:
To chime in, these templates were provided with the original eaf builder code